container/network.go

```diff
@@ -34,7 +34,10 @@ func (c *Container) CreateNetwork(namespace []string) (id string, err error) {
 }
 
 // DeleteNetwork deletes a Docker Network associated with a namespace.
-func (c *Container) DeleteNetwork(namespace []string) error {
+// event parameter can be "destroy" or "remove". If the network was used by a service, the event to use is "destroy". If the network has not been used, the event is "remove".
+// Remove removes the reference from Docker to the network.
+// Destroy removes the network from Docker active network.
+func (c *Container) DeleteNetwork(namespace []string, event string) error {
```

Comment:
Just a thought, maybe listen to both of these events inside the method. In that case, a developer doesn't need to know about internal events of docker.

Reply:
I tried and it doesn't work because the events "destroy" and "remove" have no definitive order and also "destroy" is fired only when the network has been recently (few seconds) attached to a container...
This is not a perfect solution but the perfect solution requires to reorganize the container package to match our logic (mesg's service have multiple docker's service, one network) rather than staying close to the docker logic (service, container, network). By doing this, we could extract a lot of docker related logic from the service package to the container package (like start and stop mesg's service with dependencies and network).

Reply:
Ok, got it